pkg/executor/distsql.go

```diff
@@ -643,8 +643,9 @@ func (e *IndexLookUpExecutor) startWorkers(ctx context.Context, initBatchSize in
 	// so fetching index and getting table data can run concurrently.
 	e.workerCtx, e.cancelFunc = context.WithCancel(ctx)
 	e.pool = &workerPool{
-		TolerablePendingTasks: 1,
-		MaxWorkers:            int32(max(1, e.indexLookupConcurrency)),
+		spawn: func(workers, tasks uint32) bool {
```

Comment:
The `spawn` usually indicates start a thread or some future work, how about naming it `checkSpawn` or `needSpawn`?